invertedai/utils.py

```diff
@@ -563,6 +575,211 @@ def inside_fov(center: Point, initialize_fov: float, point: Point) -> bool:
+
+    conditional_agent_recurrent_state_dict = {}
+    for agent in conditional_agent_states[-1]:
+        conditional_agent_recurrent_state_dict[(agent.center.x,agent.center.y)] = None
+
+    for i, area_center in enumerate(tmap(
+        Point.fromlist, 
+        centers, 
+        total=len(centers),
+        desc=f"Initializing {location.split(':')[1]}"
+    )):
+
+        conditional_agent = list(filter(
+            lambda x: inside_fov(center=area_center, initialize_fov=initialize_fov, point=x[0].center), 
+            zip(agent_states,agent_attributes,agent_rs)
+        ))
+
+        remaining_agents = list(filter(
+            lambda x: not inside_fov(center=area_center, initialize_fov=initialize_fov, point=x[0].center), 
+            zip(agent_states,agent_attributes,agent_rs)
+        ))
+
+        conditional_agent_specified = list(filter(
+            lambda x: inside_fov(center=area_center, initialize_fov=initialize_fov, point=x[0].center), 
+            zip(conditional_agent_states[-1],conditional_agent_attributes)
+        ))
+
+        con_agent_state = [x[0] for x in conditional_agent]
+        con_agent_attrs = [x[1] for x in conditional_agent]
+        con_agent_rs = [x[2] for x in conditional_agent]
+        remaining_agents_states = [x[0] for x in remaining_agents]
+        remaining_agents_attrs = [x[1] for x in remaining_agents]
+        remaining_agents_rs = [x[2] for x in remaining_agents]
+
+        states_history_specified_area = [x[0] for x in conditional_agent_specified]
+        states_history = states_history_specified_area + con_agent_state
+        states_history = [states_history] if len(states_history) > 0 else None
+
+        agent_attributes_specified_area = [x[1] for x in conditional_agent_specified]
+        agent_attributes_con = agent_attributes_specified_area + con_agent_attrs
+        agent_attributes_con = agent_attributes_con if len(agent_attributes_con) > 0 else None
+
+        agent_density_area_center = agent_density_list[i]
+
+        # Check if this initialization call can be skipped
+        if len(agent_attributes_specified_area) < 1:
+            # Only if there are no conditional agents in the area can it be skipped.
+            if agent_density_area_center < 1:
+                #Skip if no agents are requested (e.g. in areas with no drivable surfaces)
+                continue
+            elif len(con_agent_state) >= agent_density_area_center:
+                #Skip if the calculated number of agents has been satisfied or surpassed
+                continue
+
+        for _ in range(1):
+            try:
+                # Initialize simulation with an API call
+                response = iai.initialize(
+                    location=location,
+                    states_history=states_history,
+                    agent_attributes=agent_attributes_con,
+                    agent_count=agent_density_area_center,
```

Comment:
This argument is deprecated in initialize, so it would be better to pad `agent_attributes` instead. That would also avoid the confusing case where `agent_count` is smaller than the number of conditional agents passed in.